### PR TITLE
Adding types definitions to tabs components

### PR DESCRIPTION
**Pull request by kamilmateusiak**

### CODE REVIEW

File: src/components/Tab/Tab.js [hunk 1]
```diff
@@ -28,14 +44,19 @@ const Tab = ({ onSelect, href, isSelected, children, description }) => {
 
 Tab.propTypes = {
   children: PropTypes.node.isRequired,
+  className: PropTypes.string,
   onSelect: PropTypes.func,
```
Comment (Aideann) [inline src/components/Tab/Tab.js]: should be required? I think so

File: src/components/Tab/Tab.js [hunk 2]
```diff
@@ -28,14 +44,19 @@ const Tab = ({ onSelect, href, isSelected, children, description }) => {
...
-  description: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  onClick: PropTypes.func,
```
Comment (Aideann) [inline src/components/Tab/Tab.js]: What is the `onClick` purpose? is `onSelect` not enough?

File: src/components/Tab/Tab.js (hunk 1 above)
Comment (kamilmateusiak) [reply src/components/Tab/Tab.js]: Ok, what to do... I need to extend ButtonHTMLAttributes/AnchorHTMLAttributes where onSelect is already declared. I think about removing it from here cause it should not be the main handler.

File: src/components/Tab/Tab.js (hunk 2 above)
Comment (kamilmateusiak) [reply src/components/Tab/Tab.js]: I think`onClick` should be the main function to handle tabs.

File: src/components/Tab/Tab.js (hunk 1 above)
Comment (kamilmateusiak) [reply src/components/Tab/Tab.js]: Also It's not required 'cause `Tab` can be rendered as anchor element (there are some use cases in AA)

**Comment by dprzybylek** [comment]
@kamilmateusiak exports are sorted now, that's why you have some conflicts 😅

File: src/components/Tab/Tab.js (hunk 1 above)
Comment (Aideann) [reply src/components/Tab/Tab.js]: Ok then remove onSelect and leave only onClick ;) that will resolve every problem here

File: src/components/Tab/Tab.js (hunk 2 above)
Comment (Aideann) [reply src/components/Tab/Tab.js]: So remove onSelect ;d

Comment (kamilmateusiak) [reply src/components/Tab/Tab.js]: already did it ;)

**Pull request merged**